packages/menu/src/styles.tsx

```diff
@@ -43,28 +53,52 @@ export const Item = styled.div<ItemProps>`
     color: ${_ => _.theme.colors.text.intense};
   }
 `
-export const Text = styled(PureText)`
-  flex: 1;
-  margin-left: ${_ => _.theme.spacing.sm};
-  margin-right: ${_ => _.theme.spacing.sm};
-`
 export const SecondaryText = styled(Text)`
   margin-right: ${_ => _.theme.spacing.sm};
 `
 interface GroupProps {
   separated?: boolean
+  invalid?: boolean
 }
 export const Group = styled.div<GroupProps>`
   padding-top: ${_ => _.theme.spacing.xxs};
   padding-bottom: ${_ => _.theme.spacing.xxs};
 
-  ${Item} + ${Item} {
-    ${_ => ({
-      boxShadow: _.separated
-        ? `0 -1px 0 0 ${_.theme.colors.border.default}`
-        : undefined
-    })}
-  }
+  ${_ =>
+    _.separated && {
+      '&:hover': {
+        [`${Item}:not([data-isselectable='false'])`]: {
+          [Text]: {
+            transition: 'color 0.25s ease',
```

Comment:
Please note that transition works only when mouse is entering the field, not the other way - in order to fix it, move the attribute out of `&:hover` scope